### PR TITLE
feat: add SoftAdaptAdaptiveLoss and ReLoBRaLoAdaptiveLoss

### PR DESCRIPTION
**Pull request by jitendravjh**
## Summary

Adds two new adaptive loss weighting strategies for PINN training, requested
in issues #499, #500, #501, #446.

### `SoftAdaptAdaptiveLoss`

Based on [Heydari et al. (2019)](https://arxiv.org/abs/1912.12355).

Reweights loss components by softmax over their normalised rate of change:

```rate_i(t) = (L_i(t) - L_i(t-1)) / (L_i(t-1) + ε) λ_i(t) = softmax(α · rate(t)) × N```

No gradient computations — cheaper than `GradientScaleAdaptiveLoss`.

### `ReLoBRaLoAdaptiveLoss`

Based on [Bischof & Kraus (2021)](https://arxiv.org/abs/2110.09813).

Compares current losses to a randomly chosen past reference (initial or
previous step, controlled by `β`):

```ρ ~ Bernoulli(β) t₀ = ρ · t_prev + (1 - ρ) · t_init λ_i = softmax(α · L_i(t) / (L_i(t₀) + ε)) × N```

More stable than incremental methods for stiff PDEs.

## Usage

```julia
# SoftAdapt — reweight every 10 iterations
disc = PhysicsInformedNN(chain, strategy;
    adaptive_loss = SoftAdaptAdaptiveLoss(10))

# ReLoBRaLo — reweight every 10 iterations, β=0.9 (default)
disc = PhysicsInformedNN(chain, strategy;
    adaptive_loss = ReLoBRaLoAdaptiveLoss(10))
```
## API Compatibility
Both follow the exact same pattern as existing GradientScaleAdaptiveLoss and MiniMaxAdaptiveLoss in ```src/adaptive_losses.jl```
New generate_adaptive_loss_function dispatches — no changes to ```discretize.jl``` or the training loop
Exported from the module alongside existing adaptive loss types

### Locally verified:

Construction with defaults and custom kwargs
Integration with PhysicsInformedNN(..., adaptive_loss = ...)
discretize() on a 1D advection PDE with both methods
All existing package tests pass (no breaking changes)

## References
Heydari et al. (2019). SoftAdapt. [arXiv:1912.12355](https://arxiv.org/abs/1912.12355)
Bischof & Kraus (2021). ReLoBRaLo. [arXiv:2110.09813](https://arxiv.org/abs/2110.09813)

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
These should be added to the docs?

**Comment by jitendravjh** [comment]
Added both types to `docs/src/manual/adaptive_losses.md` in the `@docs` block. Documenter will pull the full docstrings (algorithm, kwargs, paper refs) automatically. Let me know if anything else is needed before merge.

**Review by ChrisRackauckas — APPROVED** [review]
